griptape/tasks/prompt_task.py

```diff
@@ -93,12 +97,24 @@ def after_run(self) -> None:
         super().after_run()
 
         logger.info("%s %s\nOutput: %s", self.__class__.__name__, self.id, self.output.to_text())
+        if self.conversation_memory is not None and self.output is not None:
+            run = Run(input=self.input, output=self.output)
+
+            self.conversation_memory.add_run(run)
 
     def try_run(self) -> BaseArtifact:
         message = self.prompt_driver.run(self.prompt_stack)
 
         return message.to_artifact()
 
+    def preprocess(self, structure: Structure) -> BaseTask:
+        super().preprocess(structure)
+
+        if self.conversation_memory is None and structure.conversation_memory is not None:
+            self.conversation_memory = structure.conversation_memory
```

Comment:
Won't this prevent the user from forcing task `conversation_memory` to `None`?

Reply:
Ah...right. This is where python's lack of a [sentinel value](https://peps.python.org/pep-0661/) is particularly challenging.

We need to differentiate between an unset value and an explicitly `None` value.

Reply:
[can we steal `attrs.NOTHING`?](https://www.attrs.org/en/stable/api.html#attrs.NOTHING)

Reply:
Oh nice, yes that's exactly what we need.